src/sdk/config_fs.js

```diff
@@ -1182,15 +1182,31 @@ class ConfigFS {
         }
         const running_code_config_dir_version = this.config_dir_version;
         const system_config_dir_version = system_data.config_directory.config_dir_version;
+        const ver_comparison_err = this.compare_host_and_config_dir_version(running_code_config_dir_version, system_config_dir_version);
+        if (ver_comparison_err !== undefined) {
+            throw new RpcError('CONFIG_DIR_VERSION_MISMATCH', ver_comparison_err);
+        }
+    }
+
+    /**
+     * compare_host_and_config_dir_version compares the version of the config dir in the system.json file 
+     * with the config dir version of the running host
```

Comment:
I would expect a "compare" function to return -1, 0, or 1, and in this case we return strings for errors and undefined for a match, so I think we can write it in the JSDoc so it will be clearer.